test/client/docs_test.rb

```diff
@@ -720,6 +720,60 @@
     assert_equal 1, response3["total"]
   end
 
+  it "accepts a type param and does not throw an error for ES7" do
+    if !$client.version_support.es_version_7_plus?
+      skip "This test is only needed for ES 7 onwards"
+    end
+
+    h = @docs.index \
+      _id: 1,
+      _type: "book",
+      title: "Book 1 by author 1",
+      author: "Author1"
+
+    assert_created h
+    assert_equal "1", h["_id"]
+
+    response1 = @docs.get(id: 1, type: "book")
+
+    assert_equal "1", response1["_id"]
+
+    @docs.update(document_wrapper("book", {
+      _id: "1",
+      doc: { author: "Author1.1" }
+    }))
+    response2 = @docs.get(id: "1", type: "book")
+
+    assert_equal "Author1.1", response2["_source"]["author"]
+
+    h = @docs.index \
+    _id: 2,
+    _type: "book",
+    title: "Book 2 by author 2",
+    author: "Author2"
+
+    assert_created h
+    assert_equal "2", h["_id"]
+
+    h = @docs.multi_get({ids: [2, 1]}, type: "book")
+    authors = h["docs"].map { |d| d["_source"]["author"] }
+
+    assert_equal %w[Author2 Author1.1], authors
+
+    h = @docs.delete id: 3, type: "book"
+
+    refute @docs.exists?(id: "3", type: "book")
```

Comment:
Just checking, did you want to add a doc with an `id` of 3 before these lines? Otherwise the `refute` is slightly misleading since it would pass for (almost) any values of `id` and `type`, even those not matching the `delete` (since `delete` isn't deleting an existing doc).

Reply:
Yup, my bad. The refute line was to ensure that the `delete` worked correctly, and also to test if `.exists?` works with a type.
I moved around a bunch of code and removed the indexing code for id = 3 by mistake. Thanks!